webapp/src/js/controllers/user-language-modal.js

```diff
@@ -44,10 +44,8 @@ angular.module('inboxControllers').controller('UserLanguageModalCtrl',
     };
 
     $scope.submit = function() {
-      if (!$scope.selectedLanguage) {
-        var err = new Error('No language selected');
-        $log.error(err);
-        return $q.reject(err);
+      if (typeof $scope.selectedLanguage === 'undefined') {
+        $scope.changeLanguage('en');
```

Comment:
This should default to the application default language not 'en'. Use the Settings service and get the `locale` property.

Also, if they're submitting it we should update the user so they don't get prompted next time, so call UpdateUser with the app default.